src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -897,8 +897,20 @@ public boolean isComplete() {
         // TODO: if program terminates with exception, we need to record it
         // TODO: in the error case, we have to close all the open nodes
         FlowNode head = new FlowEndNode(this, iotaStr(), (FlowStartNode)startNodes.pop(), result, getCurrentHeads().toArray(new FlowNode[0]));
-        if (outcome.isFailure())
-            head.addAction(new ErrorAction(outcome.getAbnormal()));
+        if (outcome.isFailure()) {
+            try {
+                head.addAction(new ErrorAction(outcome.getAbnormal()));
+            } catch (Exception e) {
```

Comment:
Possibly useful for corner cases where saving the `FlowNode` genuinely fails for legitimate reasons, like disk I/O problems, though I am not sure we can do much to recover in those cases anyway.

The diagnosis in this issue looks right, but the patch is much too much work, and actually loses the information in the original exception. All you needed to do was add another blacklist to `ErrorAction.isUnserializableException`. Given the Remoting blacklist pattern, and possible future changes to that, perhaps we should just say

```java
private boolean isUnserializableException(Throwable error) {
    try {
        ClassFilter.DEFAULT.check(error.getClass());
    } catch (SecurityException x) {
        return true; // e.g. PowerAssertionError
    }
    return error instanceof MultipleCompilationErrorsException ||
           error instanceof MissingMethodException;
}
```